gradient_estimators/random_gradient_estimator.py

```diff
@@ -237,15 +267,17 @@ def revert_model_given_seed_and_grad(
         iteration_seeds: Sequence[int],
         iteration_grad_scalar: Sequence[torch.Tensor],
     ) -> None:
+        # TODO: Support sgd_only_no_optim case.
+        assert not self.sgd_only_no_optim
         assert len(iteration_seeds) == len(iteration_grad_scalar)
         try:
             assert isinstance(optimizer, torch.optim.SGD) and optimizer.defaults["momentum"] == 0
         except AssertionError:
             raise Exception("Revert only supports SGD without momentum")
 
         lr, weight_decay = optimizer.defaults["lr"], optimizer.defaults["weight_decay"]
-        optimizer.zero_grad()
         for one_update_seed, one_update_grad_dirs in zip(iteration_seeds, iteration_grad_scalar):
+            optimizer.zero_grad()
             if self.paramwise_perturb:
```

Comment:
Was previous code wrong? Or this is just for memory improvement

Reply:
after second thoughts, I feel it is better to delete them and add a comment